src/main/java/com/google/cloud/resourcemanager/Lien.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.resourcemanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.api.gax.paging.Page;
+import com.google.cloud.resourcemanager.ResourceManager.LienListOption;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * A Google Cloud Resource Manager lien object.
+ *
+ * <p>A Lien represents an encumbrance on the actions that can be performed on a resource. This is
+ * an immutable class. Methods that change or update the underlying Lien information return a new
+ * Lien instance. {@code Lien} adds a layer of service-related functionality over {@link LienInfo}.
+ */
+public class Lien extends LienInfo {
+  private static final long serialVersionUID = 6767630161335155133L;
```

Comment:
are we really using object serialization here?

Reply:
done